### PR TITLE
Add New Telemetry Values

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This path adds the following values to the metadata header when requesting latest charm info from the store:
- `series`
- `arch`
- `is_controller`

## QA steps

Edit the value of `CharmRevisionUpdateInterval` in _/home/joseph/go/src/github.com/juju/juju/cmd/jujud/agent/machine.go_ so that we don't have to wait a day for metrics to be sent - a few minutes should be sufficient.

Build this patch.

Run these steps for providers LXD and another that supports "hardware characteristics" such as MAAS. This tests for both the absence and presence of metadata for architecture.

- Bootstrap and add a unit.
- `juju debug-log -m controller`.
- Check that the calls to `CharmRevisionUpdater` calls succeed without error.

Note that appearance of the actual metrics in _juju-kpi_ lags by a couple of days, so we will have to wait to check actual handling by the charm-store (unless there is another way I am not aware of).

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by mitechie** [comment]
instead of is_controller can we track the controller_uuid value instead?

**Comment by mitechie** [comment]
ignore me, I should look at the code vs just the description.

File: apiserver/facades/controller/charmrevisionupdater/updater_test.go [hunk 1]
```diff
@@ -185,16 +185,18 @@ func (s *charmVersionSuite) TestJujuMetadataHeaderIsSent(c *gc.C) {
...
 		"cloud=" + model.Cloud(),
 		"cloud_region=" + model.CloudRegion(),
-		"provider=" + cloud.Type,
+		"controller_uuid=" + s.State.ControllerUUID(),
 		"controller_version=" + version.Current.String(),
+		"environment_uuid=" + model.UUID(),
+		"is_controller=true",
+		"model_uuid=" + model.UUID(),
+		"provider=" + cloud.Type,
+		"series=quantal",
```
Comment (mitechie) [inline apiserver/facades/controller/charmrevisionupdater/updater_test.go]: can we verify that multi-series charms don't behave in any odd way and I'm also curious with the ability to have a multi-series subordinate related to different series in the same model how that would effect this. 

If the data's not perfect that's ok but I want to make sure nothing goes boom.

Comment (manadart) [reply apiserver/facades/controller/charmrevisionupdater/updater_test.go]: The series is being retrieved from the application.
The value is compared against the charm in (at least) `application.SetSeries` and `application.SetCharm`, accommodating things like `ForceSeries` and the list of supported series in charm metadata.

So it is singular in this context and tells _what the application is doing with the charm_.

I just teased this out as best as I could glean it, so let me know if this approach is incorrect.

**Comment by manadart** [comment]
[Add "series" and "architecture" to the charm metadata callback.](https://trello.com/c/lCBEf5jj/612-add-series-and-architecture-to-the-charm-metadata-callback)

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**